homeassistant/components/nissan_leaf/sensor.py

```diff
@@ -52,7 +52,8 @@ def device_class(self):
     @property
     def native_value(self):
         """Battery state percentage."""
-        return round(self.car.data[DATA_BATTERY])
+        if self.car.data[DATA_BATTERY]:
+            return round(self.car.data[DATA_BATTERY])
```

Comment:
Make sure to return a value explicitly in all cases if we return a value anywhere.
```suggestion
            return round(self.car.data[DATA_BATTERY])
        return None
```

Reply:
Done. Good catch, thanks.